### PR TITLE
WIP: initial testing checkin

### PR DESCRIPTION
**Pull request by dave3d**

### CODE REVIEW

File: .github/workflows/python-app.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
```
Comment (zivy) [inline .github/workflows/python-app.yml:18]: checkout is at v3

File: .github/workflows/python-app.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    - name: Set up Python 3.9
+      uses: actions/setup-python@v2
```
Comment (zivy) [inline .github/workflows/python-app.yml:20]: setup-python is at v4

File: .github/workflows/python-app.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+    - name: Lint with flake8
+      run: |
+        # stop the build if there are Python syntax errors or undefined names
+        flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+        # exit-zero treats all errors as warnings.
+        flake8 . --count --exit-zero --max-complexity=10 --statistics
+    - name: Run tests
+      run: |
+        sh run_tests.sh
```
Comment (zivy) [inline .github/workflows/python-app.yml:36]: Is there a need for the additional shell script? Why not run the unittest directly?

**Review by zivy — CHANGES_REQUESTED** [review]

File: .github/workflows/python-app.yml (hunk 3 above)
Comment (dave3d) [reply .github/workflows/python-app.yml:36]: yeah, probably not.  I like to have simple run scripts around so I don't have to remember exactly how to run things.

File: .github/workflows/python-app.yml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    - uses: actions/checkout@v2
+    - name: Set up Python 3.9
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.9
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install flake8
+        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
```
Comment (blowekamp) [inline .github/workflows/python-app.yml:27]: Use pip to install the code spell package: `pip install -e .`

File: .github/workflows/python-app.yml (hunk 1 above)
Comment (dave3d) [reply .github/workflows/python-app.yml]: done

File: .github/workflows/python-app.yml [hunk 5]
```diff
@@ -0,0 +1,37 @@
+# This workflow will install Python dependencies, run tests and lint with a single version of Python
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Python application
```
Comment (zivy) [inline .github/workflows/python-app.yml:4]: Possibly change the name to indicate that it is testing. 

The workflow name is what appears in the badge. For example see this [raw README](https://raw.githubusercontent.com/niaid/imaris_extensions/main/README.md) and [its rendering](https://github.com/niaid/imaris_extensions/blob/main/README.md). The text in the badge is taken from [this workflow](https://github.com/niaid/imaris_extensions/blob/44c587f043bb02bae05029c97f334c7322e4eaa1/.github/workflows/main.yml#L1). Feel free to ignore as there is no badge in the README, so no-one will see the workflow name there. Up to you.

**Review by zivy — APPROVED** [review]
See comment, feel free to ignore if you don't intend to add a badge to the README.

**Pull request merged**